[Fix] Essentials signs validity

kalebmmm opened the pull request:
Any sign in the format of &1[...] was a valid sign according to essentials, but 3rd party plugins with the same sign format would be treated equally (cancelling nearby explosions, etc)

Valid signs after this fix are:
- In the format of &1[...] as before
- Are essentials signs and enabled in the essentials config

This is in reference to #1399

Commit:
[Fix] Essentials signs validity

Any sign in the format of &1[...] was a valid sign according to essentials, but 3rd party plugins with the same sign format would be treated equally (cancelling nearby explosions, etc)

Valid signs after this fix are:
- In the format of &1[...] as before
- Are essentials signs and enabled in the essentials config

mdcfe commented:
Considering the fact that `EssentialsSign#isValidSign` is a public method, I don't think we should be changing the signature of it - other plugins might already use this method to determine their own behaviour (whether to respond to some signs for instance).

kalebmmm commented:
Yeah, I completely forgot about that.

SupaHam commented:
I'd keep the existing method as is, `@Deprecate` it and add an overload method like you've created that Essentials uses.

Commit:
Revert "[Fix] Essentials signs validity"

This reverts commit 60d95b6585184051124cd32358401653042edd18.

Commit:
[Fix - v2] Essentials signs validity

Any sign in the format of &1[...] was a valid sign according to essentials, but 3rd party plugins with the same sign format would be treated equally (cancelling nearby explosions, etc)

kalebmmm commented:
@SupaHam @md678685 Took the suggestion, 39971e5

SupaHam commented:
Thank you very much for your contribution! <3
